### ISSUE TITLE
No websites displays after compiling in current environment

### ISSUE THREAD

**Issue by GSI** [uzbl/uzbl#397]
Please paste the output of `uzbl-core --bug-info` here:

```
Commit: v0.9.1-191-g676b47e3
GTK compile: 3.22.26
GTK run: 3.22.26
WebKit compile: 2.18.3
WebKit run: 2.18.3
WebKit2: 1
libsoup compile: 2.60.2
libsoup run: 2.60.2
```

No matter which commit I compile (I tried the most recent/a few months back/beginning of year), uzbl-browser starts as expected, but hangs (idles?) at 10%.

* The target uri as provided via command line is displayed.
* PID is displayed.
* Key commands like switching to insert mode or hiding the status bar and the like work.

Note that, for my tests, I created a new user with an empty home directory, ensuring that the original **config** file is used.

Since about February I did not compile uzbl. I did now as my existing installation was failing after upgrading some packages on the system. There were error messages about **icu** which had been updated from 59.1-2 to 60.1-1. (Downgrading just this package alone would cause other errors still preventing uzbl from being run.)

Please let me know what else I may provide in order to track down the cause of this issue.

**Comment by GSI** [comment]
This is what it looks like:

**uzbl-browser**

![20171122_uzbl-browser](https://user-images.githubusercontent.com/1078037/33146036-761407dc-cfa1-11e7-9bc7-bf5ec7cb1981.png)

**uzbl-tabbed**

![20171122_uzbl-tabbed](https://user-images.githubusercontent.com/1078037/33146037-7636b9e4-cfa1-11e7-81cd-dcebb542dbc4.png)

**Comment by GSI** [comment]
Just noticed that I can also call up the bookmarks (but not surf to the corresponding sites). Any idea where I could further investigate for possible reasons?

By the way, the whole webkit versioning is really confusing. In this case, what does `WebKit2: 1` mean? "1" as in "true, you're running WebKit **2**"?

**Comment by keis** [comment]
Yes, webkit2 here is a boolean meaning you have webkit2 which is library version 4 IIRC. Confusing doesn't begin to describe it. I recently upgraded my system and see the same thing but I have yet to find the time to debug it.

**Comment by ralismark** [comment]
Happens on arch linux with both the official and AUR packages, with defaut config. Looking at the events (`uzbl-browser -vp`), only LOAD_PROGRESS 10 is sent. None of the other LOAD_* events appear.

**Comment by samjhandley** [comment]
Same for me on Arch.


`uzbl-core --bug-info
```
Commit: v0.9.1-95-g3a4c70ad
GTK compile: 3.22.7
GTK run: 3.22.26
WebKit compile: 2.14.3
WebKit run: 2.18.3
WebKit2: 1
libsoup compile: 2.56.0
libsoup run: 2.60.2
```

Lots of GTK warnings and `-vp` shows key presses are still registered but the browser appears to be locked up.

**Comment by MattDiesel** [comment]
@samjhandley The GTK warnings, if they are the same as I got, are caused by the `set uri_section ...` line in the config file. For some reason neither branch of the if statement gets used. 

I get the same issue, also on Arch, identical configuration. 

I've noticed that in `~/.cache/uzbl/` there used to be `event_daemon` files (a log file and a pid file). These aren't being created now. I don't know if that's related, but seems to be an issue with the event manager not loading. Doesn't look like a permissions issue as same behaviour running as root. 

I originally had this from community, I'm building the AUR version and also want to try modifying the build script to use the same webkit version it runs with, but building webkit takes a fair bit of time.

**Comment by greencopper** [comment]
I reported the issue on Arch on the 21. jan. 2018 too.

https://bugs.archlinux.org/task/57176

**Comment by seirios** [comment]
I'm having the same problem under Arch Linux. Hand-compiled from the latest git.

`uzbl-core --bug-info`:
```
Commit: v0.9.1-191-g676b47e3
GTK compile: 3.22.26
GTK run: 3.22.26
WebKit compile: 2.18.5
WebKit run: 2.18.5
WebKit2: 1
libsoup compile: 2.60.3
libsoup run: 2.60.3
```

@MattDiesel The files you mention are now kept in /run/user/_UID_/uzbl.

However I also think this may be related to the event manager, maybe the python version? (3.6 here)

I've tried `make test-uzbl-browser-sandbox` with the same results. On make it complains about some deprecated webkit2 functions, I wonder if that's the issue?

`make tests` says everything is OK, so maybe this is related to webkitgtk?

**UPDATE**

Recompiled webkitgtk from AUR and everything works now, under WebKit 1 however:
```
Commit: v0.9.1-47-gbdbb81ab
GTK compile: 3.22.26
GTK run: 3.22.26
WebKit compile: 2.4.11
WebKit run: 2.4.11
WebKit2: 0
libsoup compile: 2.60.3
libsoup run: 2.60.3
```

Problem is apparently with WebKit 2...

**Comment by keis** [comment]
Hey, did a quick poke around with this and it seems to be related to the javascript customizations we do. Removing `uzbl_js_init` from uzbl_init (uzbl-core.c) makes it go again. Not sure what else removing that breaks though.

**Comment by seirios** [comment]
I can confirm this.
```
Commit: v0.9.1-191-g676b47e3
GTK compile: 3.22.26
GTK run: 3.22.26
WebKit compile: 2.18.6
WebKit run: 2.18.6
WebKit2: 1
libsoup compile: 2.60.3
libsoup run: 2.60.3
```
After playing around, it seems that the mere call to JSGlobalContextCreate is enough to cause the hang. I wonder if this is a bug in javascriptcoregtk? I have tried disabling everything JS-related in uzbl, such as commands and variables, to no avail. The only thing that works is preventing the creation of the JS context, however I guess that means disabling JS altogether, which is not a good fix!

As far as I can tell, when loading an uri it doesn't get to the point of the NAVIGATION_STARTING event, associated with the signal::decide-policy of the webview. I can't make anything else of it.

**Comment by keis** [comment]
Tried asking around on the webkitgtk IRC but no response yet. Disabling that wouldn't disable javascript entirely only the bits run through the js command in the uzbl and clean contexts. That would also include indirect usage though @< expansions.

Maybe it would make sense to rip that out as a temporary measure

**Comment by GSI** [comment]
> Removing uzbl_js_init from uzbl_init (uzbl-core.c) makes it go again. Not sure what else removing that breaks though.

Confirming.

Don't know if directly related, but the first thing I noticed to be broken when running without `uzbl_js_init` is that uzbl completely hangs upon ussing "follow links" once the follow hint key is pressed. So, for example, upon entering `fl5`.

**Comment by drnlm** [comment]
One additional observation - moving uzbl_js_init to after uzbl_gui_init also side-steps the failure to load pages, although uzbl still has difficulties running javascript files from the config file.

**Comment by djeis97** [comment]
> uzbl completely hangs upon ussing "follow links" once the follow hint key is pressed. So, for example, upon entering fl5

This is inconsistent for me- sometimes follow links functions, other times it hangs. I also get random lockups, usually during a page load or during window management operations.

> One additional observation - moving uzbl_js_init to after uzbl_gui_init also side-steps the failure to load pages, although uzbl still has difficulties running javascript files from the config file.

I can confirm this- pages load, but I still get inconsistent lockups.

**Comment by msbit** [comment]
Just wanted to chime in to say it's also happening for me:

```
Commit: v0.9.1-191-g676b47e
GTK compile: 3.18.9
GTK run: 3.18.9
WebKit compile: 2.20.5
WebKit run: 2.20.5
WebKit2: 1
libsoup compile: 2.52.2
libsoup run: 2.52.2
```
